collector/smb_server_shares.go

```diff
@@ -0,0 +1,396 @@
+// +build windows
+
+package collector
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/log"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+func init() {
+	registerCollector("smb_server_shares", NewSmbServerSharesCollector, "SMB Server Shares")
+}
+
+var (
+	shareWhitelist = kingpin.Flag(
+		"collector.smb_server_shares.share-whitelist",
+		"Regexp of shares to whitelist. Share name must both match whitelist and not match blacklist to be included.",
+	).Default(".+").String()
+	shareBlacklist = kingpin.Flag(
+		"collector.smb_server_shares.share-blacklist",
+		"Regexp of shares to blacklist. Share name must both match whitelist and not match blacklist to be included.",
+	).Default("").String()
+)
+
+// A SmbServerSharesCollector is a Prometheus collector for perflib SmbServerShares metrics
+type SmbServerSharesCollector struct {
+	CurrentOpenFileCount        *prometheus.Desc
+	TotalFileOpenCount          *prometheus.Desc
+	FilesOpenedPerSec           *prometheus.Desc
+	CurrentDurableOpenFileCount *prometheus.Desc
+
+	shareWhitelistPattern *regexp.Regexp
+	shareBlacklistPattern *regexp.Regexp
+}
+
+// NewSmbServerSharesCollector ...
+func NewSmbServerSharesCollector() (Collector, error) {
+	const subsystem = "smb_server_shares"
+
+	return &SmbServerSharesCollector{
+		CurrentOpenFileCount: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "current_open_file_count"),
+			"The number of file handles that are currently open in this share (SmbServerShares.CurrentOpenFileCount)",
+			[]string{"share"},
+			nil,
+		),
+		TotalFileOpenCount: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "file_open_count"),
+			"The number of files that have been opened by the SMB File Server on behalf of its clients on this share since the server started. (SmbServerShares.TotalFileOpenCount)",
+			[]string{"share"},
+			nil,
+		),
+		FilesOpenedPerSec: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "files_opened_total"),
```

Comment:
Isn't this the same as the above metric?

Reply:
hah, no idea... I did grab what I could out of perflib